tests/func/test_import_url.py

```diff
@@ -50,16 +55,26 @@ def setUp(self):
             fobj.write("content")
 
     def test(self):
-        remove_outs_call_counter = spy(dvc.stage.Stage.remove_outs)
-        with patch.object(
-            dvc.stage.Stage, "remove_outs", remove_outs_call_counter
-        ):
-            ret = main(["import-url", self.external_source])
-            self.assertEqual(0, ret)
+        remove_outs_call_counter = self.mocker.spy(
+            dvc.stage.Stage, "remove_outs"
+        )
+        ret = main(["import-url", self.external_source])
+        self.assertEqual(0, ret)
 
         self.assertEqual(1, remove_outs_call_counter.mock.call_count)
 
 
+def test_should_remove_outs_before_import(mocker, dvc, tmp_path_factory):
+    import_dir = TmpDir(tmp_path_factory.mktemp("import"))
```

Comment:
We have `erepo_dir` fixture for such things:
`with erepo_dir.chdir():
    erepo_dir.dvc_gen("foo": "foo content")`

Reply:
Also, this test does not pass for py3.5, need to wrap `tmp_path_factory.mktemp` in `fspath`.

Reply:
Makes sense.